### DISCUSSION TITLE
关于对lua代码进行加密问题

### DISCUSSION THREAD

**Discussion by semgilo** [cloudwu/skynet#1548] [General]
想对lua代码先进加密，想把它加在c部分，又不方便直接加到lua库，有没有哪个位置比较适合处理这个buffer的地方

**Comment by hanxi** [comment]
你要么看看别人是怎么加密的？

https://github.com/viruscamp/luadec/wiki/Links-Crack

**Reply by semgilo** [reply]
感谢，我看看

**Comment by Gowa2017** [comment]
服务端有加密的必要么。。要不你编译成字节码 luac 用着。否则你就要在加载文件前做，一些工作了。

涵曦 ***@***.***> 于 2022年3月7日周一 10:01写道：

**Reply by semgilo** [reply]
公司有安全需求，那就是有必要的

**Comment by lsx1994** [comment]
服务端没必要加密吧， 是担心自己码的框架被新招的小弟拿去用么？

**Reply by semgilo** [reply]
公司要求要有的

**Reply by lsx1994** [reply]
那你可以试试修改一下luac加密的源码， 加一点异或运算。参考 https://blog.csdn.net/mycwq/article/details/47406337

**Comment by icewater2018** [comment]
请问最后采用了哪些方式加密，能分享下吗？
